File: Data Structures & Algorithms/depth-of-binary-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        
        def helper(node, depth):

            if not node or (node.left is None and node.right is None):
                return depth
            elif node.right and node.left:
                return max(helper(node.right, depth + 1), helper(node.left, depth + 1))
            elif node.right:
                return helper(node.right, depth + 1)
            return helper(node.left, depth + 1)

        if not root:
            return 0

        return helper(root, 1)
        
            

            

File: Data Structures & Algorithms/merge-two-binary-trees/submission-1.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {

private:
    TreeNode* dfs(TreeNode* root1, TreeNode* root2){
        if (root1 == nullptr){
            return root2;
        }
        if (root2 == nullptr){
            return root1;
        }
        
        root1->val = root1->val + root2->val;

        root1->left = dfs(root1->left, root2->left);
        root1->right = dfs(root1->right, root2->right);

        return root1;
    }


public:
    TreeNode* mergeTrees(TreeNode* root1, TreeNode* root2) {
        return dfs(root1, root2);
    }
};